app/src/main/java/ch/epfl/culturequest/social/ScanBadge.java

```diff
@@ -3,10 +3,32 @@
 import ch.epfl.culturequest.R;
 
 public class ScanBadge {
+    public interface Badge {
+
+        int getBadge();
+
+        static Badge identifyPlace(String s) {
+            Country country = Country.fromString(s);
+            if (country != Country.OTHER) {
+                return country;
+            }
+            City city = City.fromString(s);
+            if (city != City.OTHER) {
+                return city;
+            }
+            Museum museum = Museum.fromString(s);
+            if (museum != Museum.OTHER) {
+                return museum;
+            }
+            return Country.OTHER;
```

Comment:
How do you make the difference between a country, a city and a museum unknown, because right now you only get a country everytime. However, they got different default images to represent each type.

Reply:
The thing is we can't really know which one it is and I thought country was more general

Reply:
You can make an object containing a name and a type, that you would store in your hashmap, so that you call the right method in the interface

Reply:
I went with another way to solve the problem:
I will only display the badges that are not OTHER  has it does not really make sense but it will still be in the database in case we add new badges in the future

Reply:
Why not, maybe we will se later if we add the possibility to see the others